chroma-common/blockdevices/blockdevice_zfs.py

```diff
@@ -241,33 +241,39 @@ def filesystem_info(self):
 
         :return: message describing zfs type and datasets found, None otherwise
         """
-        self._assert_zpool('filesystem_info')
+        try:
+            self._assert_zpool('filesystem_info')
+        except NotZpoolException:
+            blockdevice = BlockDevice(self._supported_device_types[0], self._device_path.split('/')[0])
 
-        with ZfsDevice(self._device_path, False):
-            try:
-                device_names = shell.Shell.try_run(['zfs', 'list', '-H', '-o', 'name', '-r', self._device_path]).split('\n')
+            return blockdevice.filesystem_info
+        else:
+            with ZfsDevice(self._device_path, False):
+                try:
+                    device_names = shell.Shell.try_run(['zfs', 'list', '-H', '-o', 'name', '-r', self._device_path]).split('\n')
 
-                datasets = [line.split('/', 1)[1] for line in device_names if '/' in line]
+                    datasets = [line.split('/', 1)[1] for line in device_names if '/' in line]
 
-                if datasets:
-                    return "Dataset%s '%s' found on zpool '%s'" % ('s' if (len(datasets) > 1) else '',
-                                                                   ','.join(datasets),
-                                                                   self._device_path)
-                return None
-            except OSError:                             # zfs not found
-                return "Unable to execute commands, check zfs is installed."
-            except shell.Shell.CommandExecutionError as e:    # no zpool 'self._device_path' found
-                return str(e)
+                    if datasets:
+                        return "Dataset%s '%s' found on zpool '%s'" % ('s' if (len(datasets) > 1) else '',
+                                                                       ','.join(datasets),
+                                                                       self._device_path)
+                    return None
+                except OSError:                             # zfs not found
+                    return "Unable to execute commands, check zfs is installed."
+                except shell.Shell.CommandExecutionError as e:    # no zpool 'self._device_path' found
+                    return str(e)
```

Comment:
I should add, that I think using `tunefs.lustre` is an important step because the code analysis that I have done with `mount.lustre` indicates that when we hit this error, even though the `mount.lustre` is being called on a ZFS device, what `mount.lustre` reads from the device makes it believe that it's not been formatted as a ZFS device.